portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,17 +60,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
-    ArrayList<String> commentHistory = new ArrayList<String>();
+    ArrayList<ArrayList<String>> commentHistory = new ArrayList<ArrayList<String>>();
     for (Entity entity : results.asIterable()) {
       String name = (String) entity.getProperty("name");
       String text = (String) entity.getProperty("comment");
       String emoji = (String) entity.getProperty("emoji");
       String email = (String) entity.getProperty("email");
       long timestamp = (long) entity.getProperty("timestamp");
+      String imageUrl = (String) entity.getProperty("image");
 
       String commentText = name + " (" + email + ") said: " + text;
       if (!emoji.equals("None")) commentText += " " + emoji;
-      commentHistory.add(commentText);
+      //if (!imageUrl.equals(null)) commentText += "<img src=\"" + imageUrl + "\" />";
+      ArrayList<String> commentWithImage = new ArrayList<String>();
```

Comment:
Best practice would be to use a class with 2 fields. I'd say using a separate ArrayList for comments and imageUrls would probably be cleaner than this, though you'd have to coordinate the indexes, which is also not ideal.